client/src/components/Room.jsx

```diff
@@ -250,7 +261,8 @@ class Room extends React.Component {
 Room.propTypes = {
   params: React.PropTypes.object,
   userInstruments: React.PropTypes.func.isRequired,
-  socket: React.PropTypes.object
+  socket: React.PropTypes.object,
+  logIn: React.PropTypes.func
 };
```

Comment:
Put simply, login is  catch-all function which basically updates the page to reflect user name and instruments.

I may be able tor refactor this, to fix one of the bugs I listed on the pr.